spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerResourceActivationKeySpecTest.java

```diff
@@ -91,6 +99,139 @@ public void shouldAllowConsumerToRegisterWithActivationKeys() {
+        // sortable by id later than those in the second key in the combined list of pools
+        OwnerDTO owner = ownerClient.createOwner(Owners.random());
+
+        int consumerCount = 3;
+        int poolCount = 6;
+        List<ConsumerDTO> consumers = Collections.synchronizedList(new ArrayList<>());
+        List<PoolDTO> pools;
+        List<Thread> threads = new ArrayList<>();
+
+        // create 6 pools and 2 activation keys
+        ActivationKeyDTO key1 = ownerClient.createActivationKey(owner.getKey(),
+            ActivationKeys.random(owner));
+        ActivationKeyDTO key2 = ownerClient.createActivationKey(owner.getKey(),
+            ActivationKeys.random(owner));
+
+        IntStream.range(0, poolCount).forEach(entry -> {
+            ProductDTO prod = Products.random()
+                .addAttributesItem(ProductAttributes.MultiEntitlement.withValue("yes"));
+            prod = ownerProductApi.createProductByOwner(owner.getKey(), prod);
+            PoolDTO pool = ownerClient.createPool(owner.getKey(), Pools.random(prod).quantity(100L));
+            if (entry < 3) {
+                activationKeyApi.addPoolToKey(key1.getId(), pool.getId(), 1L);
+            }
+            else {
+                activationKeyApi.addPoolToKey(key2.getId(), pool.getId(), 1L);
+            }
+        });
+        pools = ownerClient.listOwnerPools(owner.getKey());
+        assertThat(pools).hasSize(poolCount);
+
+        // register new systems
+        List<Exception> exceptions = new ArrayList<>();
+        IntStream.range(0, consumerCount).forEach(entry -> {
+            threads.add(new Thread(() -> {
+                try {
+                    String uuid = StringUtil.random("uuid");
+                    ConsumerDTO consumer = Consumers.random(owner)
+                        .facts(Map.of("system.certificate_version", "3.3", "dmi.system.uuid", uuid));
+                    consumer = consumerClient.createConsumer(consumer, null, owner.getKey(),
+                        key2.getName() + "," + key1.getName(), true);
+                    // this ensures completion of creation before proceeding
+                    consumer = consumerClient.getConsumer(consumer.getUuid());
+                    assertThat(consumer).isNotNull();
+                    assertThat(consumer.getEntitlementCount()).isEqualTo(poolCount);
+                    consumers.add(consumer);
+                }
+                catch (Exception e) {
+                    exceptions.add(e);
+                }
+            }));
+        });
+        threads.forEach(Thread::start);
+        for (Thread thread : threads) {
+            while (thread.isAlive()) {
+                thread.join();
+            }
+        }
+        assertThat(exceptions).isEmpty();
+        assertThat(consumers).hasSize(consumerCount);
+        pools.forEach(pool -> assertThat(
+            client.pools().getPool(pool.getId(), null, null).getConsumed()).isEqualTo(consumerCount));
```

Comment:
We may want to assert that the pool we get back from `client.pools.getPool()` is not null before asserting against the consumer count.